.github/workflows/build-publish-pypi.yml

```diff
@@ -0,0 +1,63 @@
+name: Build KeplerGL Python Package
+
+on: push
+
+jobs:
+
+  build_and_publish:
+    runs-on: ubuntu-latest
+    if: "!contains(github.event.head_commit.message, '[ci skip]') && !contains(github.event.head_commit.message, '[skip ci]')"
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Use Node.js 12.x
+        uses: actions/setup-node@v2
+        with:
+          node-version: 12.x
+
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install twine virtualenv
+
+      - name: Build KeplerGL
+        env:
+          MapboxAccessTokenJupyter: ${{ secrets.MAPBOXTOKEN }}
+        run: |
+          python -m virtualenv venv
+          source venv/bin/activate
+          pip install jupyter jupyterlab jupyter-packaging
+          cd bindings/kepler.gl-jupyter
+          python setup.py sdist
+
+      - name: Test KeplerGL
+        run: |
+          source venv/bin/activate
+          pip install bindings/kepler.gl-jupyter/dist/*.tar.gz
+          if [ ! -f "venv/share/jupyter/nbextensions/keplergl-jupyter/index.js" ]; then
+            venv/bin/jupyter nbextension install --py --sys-prefix keplergl
+            venv/bin/jupyter nbextension enable --py --sys-prefix keplergl
+          fi
+          venv/bin/jupyter nbconvert --execute bindings/kepler.gl-jupyter/notebooks/DataFrame.ipynb --to python
+          python bindings/kepler.gl-jupyter/notebooks/DataFrame.py
+
+      - name: Create artifact 
+        uses: actions/upload-artifact@v2
+        with:
+          name: keplergl-pypi
+          path: bindings/kepler.gl-jupyter/dist/
+
+      - name: Publish Pypi
```

Comment:
what about the kepler.gl-jupyter npm module? should we add that to be additional publish step?

Reply:
Sorry, what is the keplergl-jupyter npm module?

Reply:
Do you mean the module under js/ directory? This yml will create a python package source tarball and publish it to https://pypi.org/project/keplergl/#files -- to replace the manual steps.

Reply:
I just checked the keplergl-jupyter depends on "kepler.gl": "^2.5.2", which should be taken care of by the Action `npmpublish.yml`. So, I think there is no additional publish step?

Reply:
for the js module inside the bindings, we used to release a separate npm module `keplergl-jupyter`. Because JupyterLab use to load it from the npm registry. Is that not needed anymore?

https://github.com/keplergl/kepler.gl/blob/master/bindings/kepler.gl-jupyter/RELEASE.md#to-release-a-new-version-of-keplergl-jupyter-on-npm

If so, we should update the `RELEASE.md` with the new steps. Especially what is needed to release a new version in conda-forge

Reply:
I see. I just did a quick test on my side and it seems building labextension using `jupyter labextension build .` doesn't need to load `keplergl-jupyter` from the npm registry (in JupyterLab3). So I think there is no need to publish it. I will double check and either update the `RELEASE.md` or add new Action to publish `keplergl-jupyter` to npm.

The conda-forge would be maintained in another repo separately. I can transfer the repo to kepler.gl org. Let me know if you want to add RELEASE.md in the conda-forge repo. 

Thanks!

Reply:
Yes, there is no need to publish `kepler-jupyter` to npm for JupyterLab3+, since the `jupyter labextension build .` command creates a prebuild extension for jupyterlab3. 

See https://jupyterlab.readthedocs.io/en/stable/extension/extension_dev.html :

> A prebuilt extension (new in JupyterLab 3.0) distributes a bundle of JavaScript code prebuilt from a source extension > that can be loaded into JupyterLab without rebuilding JupyterLab.
> In this case, the extension developer uses tools provided by JupyterLab to compile a source extension into a JavaScript bundle that includes the non-JupyterLab JavaScript dependencies, then distributes the resulting bundle in, for example, a Python pip or conda package. Installing a prebuilt extensions does not require Node.js.

>Because prebuilt extensions do not require a JupyterLab rebuild, they have a distinct advantage in multiuser systems where JupyterLab is installed at the system level. On such systems, only the system administrator has permissions to rebuild JupyterLab and install source extensions. Since prebuilt extensions can be installed at the per-user level, the per-environment level, or the system level, each user can have their own separate set of prebuilt extensions that are loaded dynamically in their browser on top of the system-wide JupyterLab.

Here is the log when install the prebuilt extension: (I renamed the version to 0.3.1, so you can see it doesn't pull from NPM registry)

Node v16.5.0

Yarn configuration loaded.
Building jupyterlab assets (development)
> /opt/homebrew/bin/npm pack /Users/xun/github/kepler.gl/bindings/kepler.gl-jupyter/js
-npm notice 
npm notice 📦  keplergl-jupyter@0.3.1
npm notice === Tarball Contents === 
npm notice 3.0kB README.md                                
npm notice 1.6kB babel/embed.js                           
npm notice 2.0kB babel/extension.js                       
npm notice 1.3kB babel/index.js                           
npm notice 4.6kB babel/keplergl-plugin.js                 
npm notice 9.7kB babel/keplergl/components/app.js         
npm notice 6.9kB babel/keplergl/components/config-panel.js
npm notice 1.8kB babel/keplergl/components/panel-header.js
npm notice 2.0kB babel/keplergl/components/root.js        
npm notice 1.6kB babel/keplergl/components/side-bar.js    
npm notice 7.4kB babel/keplergl/kepler.gl.js              
npm notice 2.4kB babel/keplergl/main.js                   
npm notice 2.8kB babel/keplergl/store.js                  
npm notice 2.8kB babel/keplergl/utils.js                  
npm notice 1.5kB babel/labplugin.js                       
npm notice 1.6kB babel/log.js                             
npm notice 6.1MB dist/index.js                            
npm notice 1.6kB lib/embed.js                             
npm notice 2.0kB lib/extension.js                         
npm notice 1.3kB lib/index.js                             
npm notice 2.6kB lib/keplergl-plugin.js                   
npm notice 5.2kB lib/keplergl/components/app.js           
npm notice 3.6kB lib/keplergl/components/config-panel.js  
npm notice 2.0kB lib/keplergl/components/panel-header.js  
npm notice 1.4kB lib/keplergl/components/root.js          
npm notice 2.0kB lib/keplergl/components/side-bar.js      
npm notice 6.1kB lib/keplergl/kepler.gl.js                
npm notice 2.0kB lib/keplergl/main.js                     
npm notice 2.4kB lib/keplergl/store.js                    
npm notice 3.0kB lib/keplergl/utils.js                    
npm notice 1.5kB lib/labplugin.js                         
npm notice 1.3kB lib/log.js                               
npm notice 2.9kB package.json                             
npm notice === Tarball Details === 
npm notice name:          keplergl-jupyter                        
npm notice version:       0.3.1                                   
npm notice filename:      keplergl-jupyter-0.3.1.tgz              
npm notice package size:  1.7 MB                                  
npm notice unpacked size: 6.2 MB                                  
npm notice shasum:        04100c514c0e7cafda7913b16d8970887a1a018d
npm notice integrity:     sha512-l4CLDVZC/xj29[...]kAhGopui8eOBw==
npm notice total files:   33                                      
npm notice 
keplergl-jupyter-0.3.1.tgz

Reply:
got it, that's nice, closer to notebook widget setup. Assume npm public is still needed jupyterLab <3. we should still create a npm package until JupyterLab 3 is universal

Reply:
I see. I just added an additional step to publish `keplergl-jupyter` on NPM. Also, a check logic has been added to publish to npm and pypi only with tag <version>-jupyter

```yml
      - name: Check Release Tag
        id: check-tag
        run: |
          if [[ ${{ github.event.ref }} =~ ^refs/tags/v[0-9]+\.[0-9]+\.[0-9]+-jupyter$ ]]; then
              echo ::set-output name=publish::true
          fi
      
      - name: Publish KeplerGL to Pypi
        if: steps.check-tag.outputs.publish == 'true'
        env:
          TWINE_USERNAME: __token__
          TWINE_PASSWORD: ${{ secrets.pypi_token }}
        run: |
          python -m twine upload bindings/kepler.gl-jupyter/dist/*.tar.gz 

      - name: Publish kepler-jupyter to NPM 
        if: steps.check-tag.outputs.publish == 'true'
        env:
          NODE_AUTH_TOKEN: ${{secrets.npm_token}}
        run: |
          cd bindings/kepler.gl-jupyter/js
          npm publish

```

 However, here are some notes:
1. This upgrade will create keplergl python package for Jupyter Notebook (all versions), and JupyterLab>=3 
2. However, one can still run `jupyter labextension install keplergl-jupyter` to (overwrite) install from NPM even with JupyterLab1/2

Reply:
@heshan0131 I just updated the documentation of README and RELEASE:

For both keplergl <=0.3.0 and keplergl > 0.3.0:
1. how to install keplergl for Jupyter Notebook
2. how to install keplergl for JupyterLab 1/2/3
3. how to install keplergl for Google Colab
4. how to use keplergl in Google Colab
5. how to set up local dev env

In RELEASE.md
1. how to release a new version for NPM and PyPi using Github Action (PyPi token needs to be setup)
2. how to release a new version for conda-forge (the keplergl-feedstock repo needs to be transferred to kepler.gl organization)

Let me know any questions and issues. Thanks!